src/hooks/useSetSentry.ts

```diff
@@ -0,0 +1,51 @@
+import * as Sentry from "@sentry/nextjs"
+import { useWalletSelector } from "@src/providers/WalletSelectorProvider"
+import { useCallback, useEffect } from "react"
+import { useConnect, useConnectorClient } from "wagmi"
+
+// We should use custom types as QueryKey from wagmi Array type unknown
+type QueryKey = [string, { chainId: number; connectorUid: string }]
+
+export const useSentrySetUser = ({ userAddress }: { userAddress?: string }) => {
+  useEffect(() => {
+    if (userAddress) {
+      Sentry.setUser({ id: userAddress })
+    }
+  }, [userAddress])
+}
+
+export const useSentrySetContextWallet = ({
+  userAddress,
+}: { userAddress?: string }) => {
+  const { selectedWalletId: nearWalletSelector } = useWalletSelector()
+  const { connectors: wagmiConnectors } = useConnect()
+  const wagmiConnectorClient = useConnectorClient()
+
+  const getWagmiConnectorName = useCallback(() => {
+    const queryKey = wagmiConnectorClient.queryKey as QueryKey
+    if (!queryKey || !queryKey[1]) {
+      return null
+    }
+    const connectorUid = queryKey[1].connectorUid
+    const wagmiConnector = wagmiConnectors.find(
+      (connector) => connector.uid === connectorUid
+    )
+    return wagmiConnector?.name
+  }, [wagmiConnectorClient, wagmiConnectors])
```

Comment:
This looks fragile. Doesn't wagmi exposes info about current connection?

Reply:
Unfortunately, no(

Reply:
Sad

Reply:
Is this not it? https://wagmi.sh/react/api/hooks/useAccount#connector

Reply:
) No, this is definitely not what we need. This status 'connecting' | 'reconnecting' | 'connected' | 'disconnected' doesn't tell you what wallet has been connected. Please, re-check, what this logic does.

Reply:
<img width="501" alt="Screenshot 2024-11-12 at 11 00 32" src="https://github.com/user-attachments/assets/31b2ac5d-0481-45b6-bea4-48295b9f107b">

Reply:
Please check it